### PR TITLE
[kube-state-metrics] Default to Recreate

### LINKED ISSUE #4095: [kube-state-metrics] Strategy RollingUpdate causes 'duplicate' metrics
### Is your feature request related to a problem ?

Metrics from KSM have labels like `instance` and `pod` that are filled by the KSM pod in question. Which means that when one uses RollingUpdate, it happens that you get duplicate metrics. For example (ommited some labels for clarity:

```
kube_deployment_status_replicas_available{instance="x:8080"} 1
kube_deployment_status_replicas_available{instance="y:8080"} 1
```

and thus any query that doesn't utilises `max by (pod and/or instance)` will provide a 'bad' result. 

I personally think the default should be `recreate`. - Yes, it _will_ provide a gap in data during the time you make a new release, however the trade off is IMHO better to have a gap which doesn't really hurt, than having actual visual 'bad' data. 

The screenshot here shows the difference

image

The spike is a restart with RollingUpdate, and the small gap is a restart with Recreate

### Describe the solution you'd like.

Make `recreate` rollout strategy the default and give guidance on how to prevent it otherwise.

### Describe alternatives you've considered.

NONE

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by wiardvanrij**
#### What this PR does / why we need it

See issue/ticket.

#### Which issue this PR fixes

- fixes #4095

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-state-metrics/templates/deployment.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ spec:
   replicas: {{ .Values.replicas }}
   {{- if not .Values.autosharding.enabled }}
   strategy:
-    type: {{ .Values.updateStrategy | default "RollingUpdate" }}
+    type: {{ .Values.updateStrategy | default "Recreate" }}
```
Comment (dotdc) [inline charts/kube-state-metrics/templates/deployment.yaml:23]: Please revert to `RollingUpdate`

File: charts/kube-state-metrics/values.yaml [hunk 2]
```diff
@@ -37,8 +37,11 @@ autosharding:
 
 replicas: 1
 
-# Change the deployment strategy when autosharding is disabled
-# updateStrategy: Recreate
+# Change the deployment strategy when autosharding is disabled.
+# The default is "Recreate" in favor of preventing two instances from running, but this might create a small gap in data.
+# When using "RollingUpdate", it's wise to use a pessimistic approach when querying for "kube-state-metrics" and perform something like this:
+# max by(namespace, pod) (kube_pod_container_status_restarts).
+# updateStrategy: RollingUpdate
```
Comment (dotdc) [inline charts/kube-state-metrics/values.yaml:44]: I would rework this paragraph taking into account that `RollingUpdate` is the default.
You can also add : 
```yaml
# ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
```

**Review by dotdc — CHANGES_REQUESTED** [review]
Like explained in https://github.com/prometheus-community/helm-charts/issues/4095, I would keep `RollingUpdate` as the default strategy.

**Comment by mrueg** [comment]
I would suggest to keep RollingUpdate as the default as well, rather than Recreating. 
If you run ksm with more than 1 replica, you have to deal with the instance in your query as well.
If recreate fails for some reason and a rollback is needed, you will observe larger gaps in your metrics.

**Review by dotdc — CHANGES_REQUESTED** [review]
@wiardvanrij Could you bump the chart to 5.15.4 so we can merge?

**Comment by dotdc** [comment]
Done in #4318

**Pull request closed without merging**